Document `Path::parent` behavior around relative paths

tbu- opened the pull request:
A relative path with just one component will return `Some("")` as its parent, which wasn't clear to me from the documentation.

The parent of `""` is `None`, which was missing from the documentation as well.

fbstj commented on library/std/src/path.rs:
```diff
@@ -2156,6 +2159,14 @@ impl Path {
+    /// assert_eq!(parent, Some(Path::new("foo")));
+    /// let grand_parent = parent.and_then(Path::parent);
+    /// assert_eq!(grand_parent, Some(Path::new("")));
+    /// let grand_grand_parent = grand_parent.and_then(Path::parent);
+    /// assert_eq!(grand_grand_parent, None);
```
```suggestion
    /// let great_grand_parent = grand_parent.and_then(Path::parent);
    /// assert_eq!(great_grand_parent, None);
```

tbu- replied:
Done.

tbu- committed:
Document `Path::parent` behavior around relative paths

A relative path with just one component will return `Some("")` as its
parent, which wasn't clear to me from the documentation.

The parent of `""` is `None`, which was missing from the documentation
as well.

Mark-Simulacrum commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 461d14724934f7a05fabfe74e8e51f801fa6993a has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 461d14724934f7a05fabfe74e8e51f801fa6993a -->
<!-- homu: {"type":"Approved","sha":"461d14724934f7a05fabfe74e8e51f801fa6993a","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
